http-client/src/main/java/io/micronaut/http/client/netty/ConnectionManager.java

```diff
@@ -302,6 +229,42 @@ private static NioEventLoopGroup createEventLoopGroup(HttpClientConfiguration co
         return group;
     }
 
+    /**
+     * For testing.
+     *
+     * @return Connected channels in all pools
+     */
+    @SuppressWarnings("unused")
+    List<Channel> getChannels() {
+        List<Channel> channels = new ArrayList<>();
```

Comment:
size the list based on `pools.values()` size?

Reply:
there are multiple channels per pool so that's not possible. but it's not performance critical anyway, the method is only used in some tests (the timeout specs)